RSS_STAT_ARRAY compilation failure

When I try to build bpf driver I get next error:

[configure-bpf] Including /usr/src/scap-7.2.1+driver/bpf//configure/RSS_STAT_ARRAY/Makefile.inc
[configure-bpf] Build output for HAS_RSS_STAT_ARRAY:
[configure-bpf] make: Entering directory '/usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY' make -C /host/usr/src/linux-headers-4.15.0-213-generic M=$PWD clang  -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -Iubuntu/include  \     -D__KERNEL__   \       \      -D__KERNEL__ \  -D__BPF_TRACING__ \     -Wno-gnu-variable-sized-type-not-at-end \       -Wno-address-of-packed-member \         -fno-jump-tables \      -fno-stack-protector \  -Wno-tautological-compare \   -Wno-unknown-attributes \       -O2 -g -emit-llvm -c /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/test.c -o /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/test.ll In file included from /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/test.c:18: /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:85:17: error: use of undeclared identifier 'TASK_COMM_LEN'    85 |         char prev_comm[TASK_COMM_LEN];       |                        ^ /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:89:17: error: use of undeclared identifier 'TASK_COMM_LEN'    89 |         char next_comm[TASK_COMM_LEN];       |                        ^ /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:98:19: error: use of undeclared identifier 'TASK_COMM_LEN'    98 |         char parent_comm[TASK_COMM_LEN];       |                          ^ /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:100:18: error: use of undeclared identifier 'TASK_COMM_LEN'   100 |         char child_comm[TASK_COMM_LEN];       |                         ^ /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/test.c:27:20: error: subscripted value is not an array, pointer, or vector    27 |         val = mm->rss_stat[0].count;       |               ~~~~~~~~~~~~^~ 5 errors generated. make[2]: *** [/usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/Makefile:33: /usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY/test.o] Error 1 make[1]: *** [Makefile:1596: _module_/usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY] Error 2 make: *** [Makefile:26: all] Error 2 make: Leaving directory '/usr/src/scap-7.2.1+driver/bpf/configure/RSS_STAT_ARRAY'


This lead to invalid logic of HAS_RSS_STAT_ARRAY check, and rss_stat read invalid values for kernels > 6.2.

From what I can see TASK_COMM_LEN only defined for modern_bpf. Can this problem be fixed somehow?

cat build.log
make: Entering directory '/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY'
make -C /host/usr/src/linux-headers-4.15.0-213-generic M=$PWD
make[1]: Entering directory '/host/usr/src/linux-headers-4.15.0-213-generic'
clang  -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -Iubuntu/include  \
        -D__KERNEL__   \
         \
        -D__KERNEL__ \
        -D__BPF_TRACING__ \
        -Wno-gnu-variable-sized-type-not-at-end \
        -Wno-address-of-packed-member \
        -fno-jump-tables \
        -fno-stack-protector \
        -Wno-tautological-compare \
        -Wno-unknown-attributes \
        -O2 -g -emit-llvm -c /usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/test.c -o /usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/test.ll
In file included from /usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/test.c:18:
/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:85:17: error: use of undeclared identifier 'TASK_COMM_LEN'
   85 |         char prev_comm[TASK_COMM_LEN];
      |                        ^
/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:89:17: error: use of undeclared identifier 'TASK_COMM_LEN'
   89 |         char next_comm[TASK_COMM_LEN];
      |                        ^
/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:98:19: error: use of undeclared identifier 'TASK_COMM_LEN'
   98 |         char parent_comm[TASK_COMM_LEN];
      |                          ^
/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/../../types.h:100:18: error: use of undeclared identifier 'TASK_COMM_LEN'
  100 |         char child_comm[TASK_COMM_LEN];
      |                         ^
/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/test.c:27:20: error: subscripted value is not an array, pointer, or vector
   27 |         val = mm->rss_stat[0].count;
      |               ~~~~~~~~~~~~^~
5 errors generated.
make[2]: *** [/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/Makefile:33: /usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY/test.o] Error 1
make[1]: *** [Makefile:1596: _module_/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY] Error 2
make[1]: Leaving directory '/host/usr/src/linux-headers-4.15.0-213-generic'
make: *** [Makefile:26: all] Error 2
make: Leaving directory '/usr/src/scap-7.2.0+driver/bpf/configure/RSS_STAT_ARRAY'

--- a/driver/bpf/probe.c
+++ b/driver/bpf/probe.c
@@ -14,7 +14,6 @@ or GPL2.txt for full copies of the license.
 #if __has_include(<asm/rwonce.h>)
 #include <asm/rwonce.h>
 #endif
-#include <linux/sched.h>
 
 #include "driver_config.h"
 #include "ppm_events_public.h"

--- a/driver/bpf/types.h
+++ b/driver/bpf/types.h
@@ -12,6 +12,8 @@ or GPL2.txt for full copies of the license.
 
 #ifdef __KERNEL__
 
+#include <linux/sched.h>  // TASK_COMM_LEN definition
+
 #define __bpf_section(NAME) __attribute__((section(NAME), used))
 
 #ifndef __always_inline

